File: NC/is-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sCounts = {}
        tCounts = {}

        for c in s:
            sCounts[c] = sCounts.get(c, 0) + 1

        for c in t:
            tCounts[c] = tCounts.get(c, 0) + 1

        if len(sCounts) != len(tCounts):
            return False

        for key in sCounts:
            if sCounts.get(key, 0) != tCounts.get(key, 0):
                return False

        return True

File: NC/is-anagram.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        unordered_map<char, int> m1, m2;

        for (char c : s)
        {
            m1[c]++;
        }
        for (char c : t)
        {
            m2[c]++;
        }

        if (m1.size() != m2.size())
        {
            return false;
        }

        for (auto i : m1)
        {
            if (i.second != m2[i.first])
            {
                return false;
            }
        }
        return true;
    }
};
